website/docs/bestpractices/analytics/spark-best-practices.md

```diff
@@ -0,0 +1,166 @@
+            * Hence this configuration is ideally recommended for small clusters or clusters with very low pod churn. 
+### CoreDNS Recommendations
+    * VPC resolver has a hard cap of 1024 packets per second for each ENI. By default, an EKS cluster runs two replicas of CoreDNS. Anything beyond 1024 packets per second will be throttled, and will result in `unknownHostException`. 
+    * While running Spark applications on Kubernetes, there could be scenarios where executors might perform significant number of external DNS lookups. It could be that Spark application, often needs to communicate with external services or databases, for data ingestion, processing, or need to reach external shuffle service. 
+    * To address the scalability of default coreDNS, consider implementing one of the following two options:
+        * Enable coreDNS auto-scaling.
+        * Implement Node local cache.
+    * While scaling out `CoreDNS` it is crucial to distribute replicas across different nodes. Co-locating CoreDNS on same nodes, will again end up throttling the ENI, rendering additional replicas ineffective.
+    * In order to distribute `CoreDNS` across nodes, apply node anti-affinity policy to the pods:
+
+```
+affinity:
+  podAntiAffinity:
+    requiredDuringSchedulingIgnoredDuringExecution:
+    - labelSelector:
+        matchExpressions:
+        - key: k8s-app
+          operator: In
+          values:
+          - kube-dns
+      topologyKey: kubernetes.io/hostname
+```
+
+### Reduce Inter AZ Traffic
+    * During the shuffle stage, Spark executors may need to exchange data between them. If the Pods are spread across multiple Availability Zones (AZs), this shuffle operation can turn out to be very expensive, especially on Network I/O front. Hence, for these workloads, it is recommended to colocate executors or worker pods in the same AZ. Colocating workloads in the same AZ serves two main purposes:
+        * Reduce inter-AZ traffic costs
+        * Reduce network latency between executors/Pods
+    * To have pods co-located on the same AZ, we can use podAffinity based scheduling constraints. The scheduling constraint preferredDuringSchedulingIgnoredDuringExecution can be enforced in the Pod spec. For example, in Spark we can use a custom template for our driver and executor pods:
+
+```
+spec:
+  executor:
+    affinity:
+      podAffinity:
+        preferredDuringSchedulingIgnoredDuringExecution:
+        - labelSelector:
+              matchExpressions:
+              - key: sparkoperator.k8s.io/app-name
+                operator: In
+                values:
+                - <<spark-app-name>>
+          topologyKey: topology.kubernetes.io/zone
+          ...
+```
+
+</CollapsibleContent>
+
+<CollapsibleContent header={<h2><span>Scaling Best Practices</span></h2>}>
+## Karpenter Recommendations
+    ### NodePools configuration
+        Consider creating separate NodePools for driver and executor pods. 
+        * Driver Nodepool
```

Comment:
Driver Nodepool and executor Nodepool sections should be under their own headers.